Implementing user definable behaviour for resetting shuffle, repeat and crossfade

ashenshugarRET opened the pull request:
Pull request to implement user definable behaviour for resetting shuffle, repeat and cross-fade behaviour on starting new music.
Also adding in capability of selecting Sonos favourites by name and aligning "tunein" functionality to that within hankhank10's original vinylemulator respository.

ashenshugarRET committed:
Updating process_sonos_command.js to make resetting repeat, shuffle, and crossfade user definable through usersettings.json

Added in checks to determine if repeat, shuffle, and crossfade should be reset

ashenshugarRET committed:
Upadating usersettings.json to add parameters for  repeat, shuffle, and crossfade resetting behaviour

ashenshugarRET committed:
correcting updates to process_sonos_command.js

correcting changes made to process_sonos_command.js to make resetting repeat, shuffle and crossfade user defined

ryanolf commented:
Hey there @ashenshugarRET, thanks for the pull request. Some of the tests are failing with your changes (you can run rpm test to see). The issue seems to be related to some changes to "tunein" and "favorites" urls that are unrelated to the shuffle, repeat, crossfade functionality. Perhaps you can revert those particular changes? If some of the urls are wrong, that could be  a separate pull request. Thanks!

ashenshugarRET committed:
Update process_sonos_command.js

correcting errors in code for tunein and favorites

ashenshugarRET committed:
Update usersettings.json

making reset_repeat, reset_shuffle and reset_crossfade boolean.

ashenshugarRET commented:
I've updated the code for "tunein" and "favorites" to hopefully fix the errors. Please let me know if there are any other problems.

ashenshugarRET committed:
added 'favorites' to the service not recognized error message

ashenshugarRET committed:
Update process_sonos_command.test.js

Updated process_sonos_command.test.js to matched corrected syntax in process_sonos_command.js

ashenshugarRET commented:
I think the error was caused by a mismatch between my updated process_sonos_command.js file and the process_sonos_command.test.js script with respect to the tunein service detection, I have updated the test script to hopefully correct this and to also add in the 'Favorite' service.
My updated copy of process_sonos_command.js works perfectly in my install (I simply just copied over my changes to the installed file)

ashenshugarRET committed:
Revert "Update process_sonos_command.js"

This reverts commit a6bbe8181b72391f26b7c66949d0d9cea051224e.
